packages/create-catalyst/src/utils/localization.ts

```diff
@@ -0,0 +1,51 @@
+import chalk from 'chalk';
+
+import { Https } from './https';
+
+interface Locale {
+  fallback: string | null;
+  id: string;
+  is_supported: boolean;
+  name: string;
+}
+
+interface AvailableLocalesResponse {
+  data: Locale[];
+}
+
+const isAvailableLocalesResponse = (response: unknown): response is AvailableLocalesResponse =>
+  typeof response === 'object' &&
+  response !== null &&
+  'data' in response &&
+  Array.isArray(response.data) &&
+  response.data.every(
+    (item) =>
+      typeof item === 'object' &&
+      item !== null &&
+      'id' in item &&
+      'name' in item &&
+      'fallback' in item &&
+      'is_supported' in item,
+  );
+
+export const getAvailableLocales = async (bc: Https) => {
+  const availableLocalesResponse = await bc.fetch('/v3/settings/store/available-locales');
+
+  if (!availableLocalesResponse.ok) {
+    console.error(
+      chalk.red(
+        `\nGET /v3/settings/store/available-locales failed: ${availableLocalesResponse.status} ${availableLocalesResponse.statusText}\n`,
+      ),
+    );
+    process.exit(1);
+  }
+
+  const availableLocales: unknown = await availableLocalesResponse.json();
+
+  if (!isAvailableLocalesResponse(availableLocales)) {
+    console.error(chalk.red('\nUnexpected response format from available-locales endpoint\n'));
+    process.exit(1);
```

Comment:
Can we have these utils throw an error rather than exiting the process. `process.exit` should be apart of the application flow, rather than "utilities".

Reply:
Yeah, it make sense for me 👍 
Should I catch it outside and existing the process?

Reply:
Just throw an error, the whole process should stop as I don't think we are properly wrapping the code in `try/catch`.